web/src/users/add_url.rs

```diff
@@ -0,0 +1,41 @@
+use axum::{
+    extract::{Form, State},
+    response::{IntoResponse, Redirect},
+};
+use cja::jobs::Job;
+
+use crate::{jobs::process_article::ProcessArticle, AppState};
+
+use db::{urls::add_url, users::User};
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub struct ArticleForm {
+    url: String,
+}
+
+#[axum::debug_handler(state = AppState)]
+pub async fn insert_article_handler(
+    State(state): State<AppState>,
+    user: User,
+    Form(form): Form<ArticleForm>,
+) -> impl IntoResponse {
+    let url = form.url;
+    let user_id = user.user_id;
+
+    match add_url(&state.db, &url, user_id, &true).await {
+        Ok(page) => {
+            let page_id = page.page().page_id;
+            let process_article = ProcessArticle { page_id };
+            process_article
+                .enqueue(state.clone(), "context".to_string())
+                .await
+                .unwrap();
+            Redirect::to(&format!("/article/{page_id}?flash[success]=Article added"))
+        }
+        Err(e) => {
+            eprintln!("Error adding URL: {e:?}");
+            Redirect::to("/dashboard?flash[error]=Could not add article")
```

Comment:
Ok so I think we want to change this here, and unfortunately we are going to lose this nice redirect and flash... But it will lean into the existing Error Reporting better, and we will add back in nice redirects and flash messages later I think!

So I think what we want to do here is make this function return a `WebResult<Redirect>` (or `WebResult<impl IntoResponse>` if we don't want to be specific about the redirect, will compile the same just means if we change it from a redirect we won't have to change the signature 🤷 )

And then we should be able to use `?` on the `Result`s in this function to return a generic `500` error if they fail, which also handles reporting the error with the `error!` tracing macro: https://github.com/coreyja/knowledge/blob/main/web/src/err.rs#L30

And then you don't have to do the `match` here for the error, you can just use `?` to forward that up and have that `Error` renderer I linked run!

And can we do the same for the `unwrap()` after the enqueue? I _think_ that will work for both a `miette` error and `color_eyre` error (and its on my list to fix this likely my just moving to `color_eyre` in cja soon!)

Reply:
What's the reason behind this change?